### PR TITLE
Update to pinejs 15

### PR DESCRIPTION
**Pull request by Page-**
See: https://balena.zulipchat.com/#narrow/stream/346007-balena-io.2FbalenaCloud/topic/pinejs.2015

### CODE REVIEW

File: src/features/registry/middleware.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ export const basicApiKeyAuthenticate: RequestHandler = async (
 		req.params.subject = creds.name;
 		req.params.apikey = creds.pass;
 	}
+	if (req.params.apikey === TOKEN_AUTH_BUILDER_TOKEN) {
```
Comment (thgreasi) [inline src/features/registry/middleware.ts:26]: Is the `TOKEN_AUTH_BUILDER_TOKEN` actually used any more?
I couldn't find any references outside the API :thinking:
See: https://github.com/search?type=code&auto_enroll=true&q=%22TOKEN_AUTH_BUILDER_TOKEN%22+-path%3A*.md+-path%3A%2F.versionbot%2F*+NOT+is%3Aarchived+%28org%3Abalena-io+OR+org%3Abalena-io-examples+OR+org%3Abalena-io-library+OR+org%3Abalena-io-modules+OR+org%3Abalena-os+OR+org%3Abalenablocks+OR+org%3Abalena-labs-projects+OR+org%3Abalenaltd+OR+org%3Aproduct-os%29

File: src/migrations/00082-pinejs-15.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
+-- Boolean type conversions
+
+ALTER TABLE "application type"
+ALTER COLUMN "supports web url" SET DATA TYPE BOOLEAN USING b::BOOLEAN,
```
Comment (thgreasi) [inline src/migrations/00082-pinejs-15.sql:4]:
```suggestion
ALTER COLUMN "supports web url" SET DATA TYPE BOOLEAN USING "supports web url"::BOOLEAN,
```
Applies in all conversions.

**Review by thgreasi — COMMENTED** [review]
:clap:

File: src/features/registry/middleware.ts (hunk 1 above)
Comment (Page-) [reply src/features/registry/middleware.ts:26]: It's certainly set, I don't remember what uses it but I'm also not fussed about it in terms of this PR as it's definitely not something to bundle with it

File: src/migrations/00082-pinejs-15.sql [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ALTER COLUMN "is undervolted" SET DEFAULT FALSE;
+
+ALTER TABLE "release"
+ALTER COLUMN "is invalidated" DROP DEFAULT,
+ALTER COLUMN "is passing tests" DROP DEFAULT,
+ALTER COLUMN "is invalidated" SET DATA TYPE BOOLEAN USING "is invalidated"::BOOLEAN,
+ALTER COLUMN "is passing tests" SET DATA TYPE BOOLEAN USING "is passing tests"::BOOLEAN,
+ALTER COLUMN "is invalidated" SET DEFAULT FALSE,
+ALTER COLUMN "is passing tests" SET DEFAULT TRUE;
+
```
Comment (fisehara) [inline src/migrations/00082-pinejs-15.sql:52]:
```suggestion
DROP CONSTRAINT "release$4VHBqaOH1cZHOsRGD2NvfC+SbBdzlb1ZY/X5Nf4LYrk=";
CONSTRAINT "release$4VHBqaOH1cZHOsRGD2NvfC+SbBdzlb1ZY/X5Nf4LYrk=" CHECK (NOT (
	"invalidation reason" IS NOT NULL
	AND "is invalidated" != TRUE
))
```

**Review by fisehara — COMMENTED** [review]
@Page-  ✅

File: src/migrations/00082-pinejs-15.sql [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ALTER COLUMN "is connected to vpn" SET DEFAULT FALSE,
+ALTER COLUMN "is undervolted" SET DEFAULT FALSE;
+
+ALTER TABLE "release"
+ALTER COLUMN "is invalidated" DROP DEFAULT,
+ALTER COLUMN "is passing tests" DROP DEFAULT,
+ALTER COLUMN "is invalidated" SET DATA TYPE BOOLEAN USING "is invalidated"::BOOLEAN,
+ALTER COLUMN "is passing tests" SET DATA TYPE BOOLEAN USING "is passing tests"::BOOLEAN,
+ALTER COLUMN "is invalidated" SET DEFAULT FALSE,
+ALTER COLUMN "is passing tests" SET DEFAULT TRUE;
```
[the remark this answers is not in the document]
Comment (Page-) [reply src/migrations/00082-pinejs-15.sql:51]: No, it's overridden as default true, trust me those overrides have caused me enough headaches already..

**Comment by Page-** [comment]
LGTM

**Pull request merged**